### PR TITLE
feat: Added cdn profile security policies support - `avm/res/cdn/profile`

### LINKED ISSUE #2376: [Feature Request]: Front Door Standard/Premium Support
### Description

Currently CARML is only supporting the Front Door service now known as "classic", since when the new Azure Front Door Standard/Premium was introduced in preview (but has been promoted to [GA since 29 March 2022](https://azure.microsoft.com/en-us/blog/introducing-the-new-azure-front-door-reimagined-for-modern-apps-and-content/)).

[CARML Front Door module](https://github.com/Azure/ResourceModules/tree/main/modules/Microsoft.Network/frontDoors) is currently using API Version 2020-05-01 which is [the one used for Front Door "Classic" deployment](https://learn.microsoft.com/en-us/azure/frontdoor/quickstart-create-front-door-bicep?tabs=CLI).

CARML should be updated to also support Front Door Standard/Premium deployment, as per [Microsoft Sample with the new API Version and new set of BICEP resources](https://learn.microsoft.com/en-us/azure/frontdoor/create-front-door-bicep?tabs=CLI) that need to be deployed.
Part of the code is already there in the [https://github.com/Azure/ResourceModules/tree/main/modules/Microsoft.CDN/profiles](Microsoft.CDN/profiles) modules, but some bits are not supported/allowed eg in the CDN Profile SKU Name the FrontDoor SKUs are not supported:

- 'Standard_AzureFrontDoor'
- 'Premium_AzureFrontDoor'

There is currently no issue regarding FD Standard/Premium support, that's why I have opened this new one.

### PR DESCRIPTION
**Pull request by anderseide**
## Description

Adds support for Security Policies on cdn/profile, making it possible to use the module to deploy Azure Front Door Premium with Web Application Firewall.

Fixes #2376

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.cdn.profile](https://github.com/anderseide/avm-bicep-registry-modules/actions/workflows/avm.res.cdn.profile.yml) |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by gbeaud** [comment]
Looks good to me but I would appreciate a second look from the core team, thanks! 👍

File: avm/res/cdn/profile/securityPolicies/main.bicep
Comment (AlexanderSehr) [inline avm/res/cdn/profile/securityPolicies/main.bicep:36]: Hey @anderseide,
thank your for your contribution 🚀 

@gbeaud, I hope you don't mind me chiming in. Just one suggestion if I may:

IF you happen to add the `@export()` annotation to this type, you could actually create a UDT for the `securityPolicies` in the parent too, import the type there and just reuse it. This would look similar to
https://github.com/Azure/bicep-registry-modules/blob/e006349901e74388e1623b1572bc08a4f15edf75/avm/res/document-db/database-account/main.bicep#L914-L918

like
```bicep
@description('Optional. Array of Security Policy objects (see https://learn.microsoft.com/en-us/azure/templates/microsoft.cdn/profiles/securitypolicies for details).')
param securityPolicies securityPolicyType[]?

module profile_securityPolicies 'securityPolicies/main.bicep' = [
  for (securityPolicy, index) in (securityPolicies ?? []): {
    name: '${uniqueString(deployment().name)}-Profile-SecurityPolicy-${index}'
    params: {
      name: securityPolicy.name
      profileName: profile.name
      associations: securityPolicy.associations
      wafPolicyResourceId: securityPolicy.wafPolicyResourceId
    }
    dependsOn: [
      profile_afdEndpoints
      profile_customDomains
    ]
  }
]

import { associationsType } from 'securityPolicies/main.bicep'
type securityPolicyType = {
  @description('Required. (...)')
   name: string

  @description('Required. (...)')
   associations: associationsType 

  @description('Required. (...)')
   wafPolicyResourceId: string
}
```

Comment (anderseide) [reply avm/res/cdn/profile/securityPolicies/main.bicep:36]: Good suggestion! I can update the code to use export if you like @gbeaud. I didn't use it because it's still marked as experimental, but if it has @AlexanderSehr's blessing I guess it's good enough 😃

Comment (AlexanderSehr) [reply avm/res/cdn/profile/securityPolicies/main.bicep:36]: Hey @anderseide, could you elaborate which feature is experimental? Biceo type imports from other files should be 'good to go' by now (just not from e.g., the bicep-types repository). Or are you referring to the security policies? If the case, then I'd still argue for providing the UDT - but it would get us into a whole other conversation, so I'll hold onto that for now 😄

Comment (anderseide) [reply avm/res/cdn/profile/securityPolicies/main.bicep:36]: I'm referring to the experimental flag `compileTimeImports` that must be enabled from the [Bicep config file](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-config#enable-experimental-features) as stated in the UDT [documentation here](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/user-defined-data-types#import-types-between-bicep-files). It's as far as I know it's required to be able to use `@export`. I see it's enabled for the AVM repo so maybe it's safe enough to use...

Comment (AlexanderSehr) [reply avm/res/cdn/profile/securityPolicies/main.bicep:36]: Hey @anderseide,
I can't see it 😄 
https://github.com/Azure/bicep-registry-modules/blob/e006349901e74388e1623b1572bc08a4f15edf75/bicepconfig.json#L1-L12
Neither here, nor anywhere in the repository. I assume it was since moved out of the experimental state and the docs were not correctly updated. This would correlate to the mentioned Bicep version `0.21.X` as we're now at `0.29.4` if I'm not mistaken. As far as I'm concerend ... gogogo

Comment (anderseide) [reply avm/res/cdn/profile/securityPolicies/main.bicep:36]: Uum, yeah, your are absolutely right! My bad 🤦 - I'll update the code, and I'll try to find time to send a PR for the docs as well...

Comment (anderseide) [reply avm/res/cdn/profile/securityPolicies/main.bicep:37]: Updated code - running new deployment tests now. Also created PR for azure-docs

**Comment by eriqua** [comment]
Hey @anderseide, sorry to chime in. Unfortunately meanwhile the main.json file got some conflicts. Could you kindly merge latest changes from upstream and solve them?

Other than that @gbeaud would this PR be ready for approval?

Just a last step to drive this contribution to closure 💪

**Comment by anderseide** [comment]
Hi @eriqua! Latest changes has been merged

File: avm/res/cdn/profile/main.bicep
Comment (eriqua) [inline avm/res/cdn/profile/main.bicep:52]: @anderseide bare with me, another small one from my side 😉 
Since new functionality is added, could you please raise minor version in version.json? Thanks

Comment (anderseide) [reply avm/res/cdn/profile/main.bicep:52]: Done

**Review by gbeaud — APPROVED** [review]
Thanks for the work @anderseide and for adding the new deployment example "Azure Front Door Premium" and tests along with the security policies! Thanks @eriqua as well for the support on the review 💪

Minor comment: in avm/res/cdn/profile/main.bicep there's a typo ("math" instead of "match" which ends up in the JSON and README) but I don't want you to re-run the whole deployment process for this so I'll fix it in my next PR, no worries 👍

**Pull request merged**
